geometric_kernels/kernels/feature_maps.py

```diff
@@ -75,3 +76,35 @@ def _map(X: B.Numeric) -> B.Numeric:
     _context: Dict[str, str] = {"key": key}
 
     return _map, _context
+
+
+@dispatch
+def random_phase_feature_map(
+    space: NoncompactSymmetricSpace, params, state, key, order=100
```

Comment:
It's a bit trick one. On `NoncompactSymmetricSpace` we have the kernel defined in terms of a feature map, so it kinda makes little sense to take `kernel` as input (which would require instantiating a kernel, which in turn -- would require creating a feature map, making an infinite loop). 

I opted out to just drop the `kernel` input in this case. This kinda breaks the uniformity of the interface, but since we are going to provide a "sensible defaults" feature_map/kernel generator, I think it's fine.

Reply:
@vabor112

Reply:
It's an option, yes. To discuss: maybe just drop kernel dependence in the compact case as well?